src/fsrs/impl/long_term_schduler.ts

```diff
@@ -0,0 +1,276 @@
+import { AbstractScheduler } from '../abstract_schduler'
+import { TypeConvert } from '../convert'
+import {
+  type Card,
+  type Grade,
+  Rating,
+  type RecordLogItem,
+  State,
+} from '../models'
+import type { int } from '../types'
+
+export default class LongTermScheduler extends AbstractScheduler {
+  protected override newState(grade: Grade): RecordLogItem {
+    const exist = this.next.get(grade)
+    if (exist) {
+      return exist
+    }
+
+    const first_difficulty = this.algorithm.init_difficulty(grade)
+    const first_stability = this.algorithm.init_stability(grade)
```

Comment:
Each `grade` should have its corresponding `stability` and `difficulty` set.

Reply:
Before the first learning, the card doesn't have stability and difficulty. In the first learning, each grade has its corresponding w[0:4]. What's the problem here?

Reply:
> Before the first learning, the card doesn't have stability and difficulty. In the first learning, each grade has its corresponding w[0:4]. What's the problem here?

The `preview` in `abstract_scheduler.ts` collects results for all grades [Again, Good, Hard, Easy], and after the first call to `review`, it stores the results, causing all to have `init_stability` and `init_difficulty` set according to `Rating.Again`.
https://github.com/open-spaced-repetition/ts-fsrs/blob/d43b6b006e24875550ca3ce842958598fb2c8679/src/fsrs/abstract_schduler.ts#L48-L74

Reply:
Where does the `again` come from?